kafka-connect-transforms/src/main/java/io/tabular/iceberg/connect/transforms/util/KafkaMetadataAppender.java

```diff
@@ -0,0 +1,249 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package io.tabular.iceberg.connect.transforms.util;
+
+import java.util.Map;
+import java.util.function.Function;
+import org.apache.iceberg.relocated.com.google.common.collect.Maps;
+import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.common.config.ConfigException;
+import org.apache.kafka.connect.connector.ConnectRecord;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.SchemaBuilder;
+import org.apache.kafka.connect.data.Struct;
+import org.apache.kafka.connect.sink.SinkRecord;
+import org.apache.kafka.connect.transforms.util.SimpleConfig;
+
+public abstract class KafkaMetadataAppender {
+
+  public static final String INCLUDE_KAFKA_METADATA = "kafka_metadata.include";
+  public static final String EXTERNAL_KAFKA_METADATA = "kafka_metadata.external_field";
+  public static final String KEY_METADATA_FIELD_NAME = "kafka.metadata.field_name";
+  public static final String KEY_METADATA_IS_NESTED = "kafka_metadata.nested";
+  public static final String DEFAULT_METADATA_FIELD_NAME = "_kafka_metadata";
+
+  public static final ConfigDef CONFIG_DEF =
+      new ConfigDef()
+          .define(
+              KafkaMetadataAppender.INCLUDE_KAFKA_METADATA,
+              ConfigDef.Type.BOOLEAN,
+              true,
+              ConfigDef.Importance.LOW,
+              "Include appending of Kafka metadata to SinkRecord")
+          .define(
+              KafkaMetadataAppender.KEY_METADATA_FIELD_NAME,
+              ConfigDef.Type.STRING,
+              KafkaMetadataAppender.DEFAULT_METADATA_FIELD_NAME,
+              ConfigDef.Importance.LOW,
+              "field to append Kafka metadata under")
```

Comment:
Not sure what the question is?

Reply:
Oh.  I see.  That would break things. 

It's theoretically possible to make that happen --if the existing field was a struct but there is no guarantee.

At some point users have to be responsible for their own actions.

Reply:
If it breaks things today, I'm good with it 👍 

I don't think it would hurt to support such behaviour in the future, but I'm content to wait for the feature request before doing this.